zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepoCommon.java

```diff
@@ -0,0 +1,261 @@
+package org.apache.zeppelin.notebook.repo;
+
+import com.google.common.collect.Lists;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.vfs2.FileContent;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.commons.vfs2.FileType;
+import org.apache.commons.vfs2.NameScope;
+import org.apache.commons.vfs2.VFS;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.notebook.ApplicationState;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.NoteInfo;
+import org.apache.zeppelin.notebook.Paragraph;
+import org.apache.zeppelin.scheduler.Job;
+import org.apache.zeppelin.user.AuthenticationInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NotebookRepoCommon {
```

Comment:
Any reason to create a new class instead of putting them in VFSNotebookRepo ?

Reply:
There's a lot of common functionality between `GitNotebookRepo` and `VFSNotebookRepo` one way was to refactor these to `NotebookRepoCommon` or `class GitNotebookRepo extends VFSNotebookRepo implements NotebookRepoWithVersionControl`, shall I do the later?

Reply:
The later make more sense to me.

Reply:
Sure, have reverted `NotebookRepoCommon` related changes.